SPECS/xcp-ng-release.spec

```diff
@@ -439,6 +439,27 @@ EOF
  distroverpkg=centos-release
 EOF
 
+# XCP-ng: change depmod global configuration to give priority to 'override' modules dir
+%triggerin config -- kmod
+DEPMOD_PATCH=$(cat <<'EOF'
+--- /etc/depmod.d/dist.conf.orig        2019-04-23 11:31:19.107096410 +0200
++++ /etc/depmod.d/dist.conf     2019-04-23 11:31:30.533088996 +0200
+@@ -3,4 +3,4 @@
+ #
+
+ # override default search ordering for kmod packaging
+-search updates extra built-in weak-updates
++search override updates extra built-in weak-updates
+EOF
+)
+# Do not apply patch if it was already applied
+if ! echo "$DEPMOD_PATCH" | patch --dry-run -RsN -d / -p1 >/dev/null; then
```

Comment:
This part is a bit more convoluted than the equivalent `%triggerin` sections inherited from XenServer, just above in the file. 

Differences:
* They always attempt to apply the patch, but output is redirected to /dev/null and no difference is made between hooks ignored because already applied and actual failures. So the patching may fail without anyone noticing.
* We want an error to be shown if the patch really fails to apply, so first we check if the patch was already applied and skip it if it's the case, then we do apply it and display an error message if there's something wrong.

We do not end the script with a nonzero return code, because RPM transactions don't like that. But that will give useful feedback in RPM installation logs.